pkg/cmd/cluster_bootstrap.go

```diff
@@ -92,6 +92,7 @@ func newClusterBootstrapCmd() *cobra.Command {
 			if err := cmd.ValidateRequiredFlags(); err != nil {
 				return err
 			}
+			o.kubeconfig = cmd.Flag("kubeconfig").Value.String()
```

Comment:
I am not very familiar with Cobra. This `PersistentPreRunE` is executed before `RunE`, right?
Since the evaluation of the `kubeconfig` flag was moved here, I assume the other flags are evaluated just before run is called? In that do we not need to move the other flags up here as well to ensure the desired values passed as flags are considered?

Reply:
Yes It's always executed before the `RunE`, this is why it contains the permission check since the main logic does not even start without the `permissionCheck` to succeed.
`kubeconfig` flag is registered here: https://github.com/kubernetes-sigs/controller-runtime/blob/aaaefb43f7e0e8e1b81371cc1b4705a967dfa0bc/pkg/client/config/config.go#L49-L58 and we are using this function to load the kubeconfig: 
https://github.com/kubernetes-sigs/controller-runtime/blob/aaaefb43f7e0e8e1b81371cc1b4705a967dfa0bc/pkg/client/config/config.go#L116-L136

Since this flag is not registered by Cobra I need to get it somehow to be able to work with that and since `PersistentPreRunE` always runs I load the value of the flag here.